### PR TITLE
Make RDSPostgres/Aurora DBSnapshot delete operation Idempotent

### PR DESCRIPTION
**Pull request by A-kanksh-a**
## Change Overview

This PR makes the delete operation for RDS DB snapshot Idempotent. It looks for the `DB snapshot Not Found` error in the case of PostgreSQL or `DBClusterSnapshotNotFoundFault` in the case of Aurora DBCluster and prints the message in the log. 

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

`./build/integration-test.sh RDSPostgreSQLSnap` 
[https://gist.github.com/A-kanksh-a/f56f06825b42c43653b2678bfde324b1](https://gist.github.com/A-kanksh-a/f56f06825b42c43653b2678bfde324b1)

### CODE REVIEW

File: pkg/function/delete_rds_snapshot.go [hunk 1]
```diff
@@ -67,10 +69,18 @@ func deleteRDSSnapshot(ctx context.Context, snapshotID string, profile *param.Pr
...
+		if err != nil {
+			if err, ok := err.(awserr.Error); ok {
+				switch err.Code() {
+				case awsrds.ErrCodeDBSnapshotNotFoundFault:
+					log.Info().Print("DB Snapshot already deleted", field.M{"SnapshotId": snapshotID})
+					return nil, nil
+				default:
+					return nil, errors.Wrap(err, "Failed to delete snapshot")
+				}
+			}
```
Comment (viveksinghggits) [inline pkg/function/delete_rds_snapshot.go:82]: should we do this for aurora clusters as well.

File: pkg/function/delete_rds_snapshot.go [hunk 2]
```diff
@@ -67,10 +69,18 @@ func deleteRDSSnapshot(ctx context.Context, snapshotID string, profile *param.Pr
...
 		// Delete Snapshot
 		log.Print("Deleting RDS snapshot", field.M{"SnapshotID": snapshotID})
-		if _, err := rdsCli.DeleteDBSnapshot(ctx, snapshotID); err != nil {
-			return nil, errors.Wrap(err, "Failed to delete snapshot")
+		_, err := rdsCli.DeleteDBSnapshot(ctx, snapshotID)
+		if err != nil {
+			if err, ok := err.(awserr.Error); ok {
+				switch err.Code() {
+				case awsrds.ErrCodeDBSnapshotNotFoundFault:
+					log.Info().Print("DB Snapshot already deleted", field.M{"SnapshotId": snapshotID})
```
Comment (viveksinghggits) [inline pkg/function/delete_rds_snapshot.go:77]:
```suggestion
					log.Info().Print("DB Snapshot is already deleted", field.M{"SnapshotId": snapshotID})
```

File: pkg/function/delete_rds_snapshot.go [hunk 3]
```diff
@@ -79,8 +89,17 @@ func deleteRDSSnapshot(ctx context.Context, snapshotID string, profile *param.Pr
...
 	// delete Aurora DB cluster snapshot
 	log.Print("Deleting Aurora DB cluster snapshot")
-	if _, err := rdsCli.DeleteDBClusterSnapshot(ctx, snapshotID); err != nil {
-		return nil, errors.Wrap(err, "Error deleting Aurora DB cluster snapshot")
+	_, err = rdsCli.DeleteDBClusterSnapshot(ctx, snapshotID)
+	if err != nil {
+		if err, ok := err.(awserr.Error); ok {
+			switch err.Code() {
+			case awsrds.ErrCodeDBClusterSnapshotNotFoundFault:
+				log.Info().Print("DB Cluster Snapshot already deleted", field.M{"SnapshotId": snapshotID})
```
Comment (viveksinghggits) [inline pkg/function/delete_rds_snapshot.go:97]:
```suggestion
				log.Info().Print("DB Cluster Snapshot is already deleted", field.M{"SnapshotId": snapshotID})
```

**Review by viveksinghggits — APPROVED** [review]

File: pkg/function/delete_rds_snapshot.go (hunk 2 above)
Comment (pavannd1) [reply pkg/function/delete_rds_snapshot.go:77]:
```suggestion
					log.Info().Print("Could not find matching RDS snapshot; might have been deleted previously", field.M{"SnapshotId": snapshotID})
```

File: pkg/function/delete_rds_snapshot.go (hunk 3 above)
Comment (pavannd1) [reply pkg/function/delete_rds_snapshot.go:97]:
```suggestion
				log.Info().Print("Could not find matching Aurora DB cluster snapshot; might have been deleted previously", field.M{"SnapshotId": snapshotID})
```

**Pull request merged**